core/src/avm2/vector.rs

```diff
@@ -265,21 +285,22 @@ impl<'gc> VectorStorage<'gc> {
     ) -> Result<Value<'gc>, Error<'gc>> {
         self.check_fixed(activation)?;
 
-        let unshifted = if self.storage.is_empty() {
-            None
+        if !self.storage.is_empty() {
+            Ok(self.storage.remove(0))
         } else {
-            Some(self.storage.remove(0))
-        };
-
-        match unshifted {
-            Some(v) => Ok(v),
-            None if Object::ptr_eq(self.value_type(), activation.avm2().classes().uint) => {
-                Ok(Value::Integer(0))
-            }
-            None if Object::ptr_eq(self.value_type(), activation.avm2().classes().int) => {
-                Ok(Value::Integer(0))
+            if let Some(value_type) = self.value_type() {
+                if Object::ptr_eq(value_type, activation.avm2().classes().uint) {
+                    Ok(Value::Integer(0))
+                } else if Object::ptr_eq(value_type, activation.avm2().classes().int) {
```

Comment:
Should the `int` and `uint` if-statements be meged into a single statement?

Reply:
They could, just thought it'd be slightly cleaner this way.